tool/site/client/tests/components/dispense/createDispenseNotification.spec.ts

```diff
@@ -0,0 +1,136 @@
+  dispenseFormValues.prescription.statusCode = PrescriptionStatus.PARTIALLY_DISPENSED
+
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  expect(result).toMatchSnapshot()
+})
+
+test("Produces expected result when status full dispense is done on a partial dispense", () => {
+  dispenseFormValues.lineItems[0].priorStatusCode = LineItemStatus.PARTIALLY_DISPENSED
+  dispenseFormValues.lineItems[0].statusCode = LineItemStatus.DISPENSED
+  dispenseFormValues.prescription.priorStatusCode = PrescriptionStatus.PARTIALLY_DISPENSED
+  dispenseFormValues.prescription.statusCode = PrescriptionStatus.DISPENSED
+
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  expect(result).toMatchSnapshot()
+})
+
+test("Quantity is populated correctly for full dispense", () => {
+  dispenseFormValues.lineItems[0].priorStatusCode = LineItemStatus.TO_BE_DISPENSED
+  dispenseFormValues.lineItems[0].statusCode = LineItemStatus.DISPENSED
+  dispenseFormValues.prescription.statusCode = PrescriptionStatus.DISPENSED
+
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  const resultMedicationDispense = getMedicationDispenseResources(result)
+
+  expect(resultMedicationDispense[0].quantity.value).toEqual(63)
+})
+
+test("Quantity is populated correctly for partial dispense", () => {
+  dispenseFormValues.lineItems[0].priorStatusCode = LineItemStatus.TO_BE_DISPENSED
+  dispenseFormValues.lineItems[0].statusCode = LineItemStatus.PARTIALLY_DISPENSED
+  dispenseFormValues.lineItems[0].suppliedQuantityValue = "1"
+  dispenseFormValues.prescription.statusCode = PrescriptionStatus.PARTIALLY_DISPENSED
+
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  const resultMedicationDispense = getMedicationDispenseResources(result)
+
+  expect(resultMedicationDispense[0].quantity.value).toEqual(1)
+})
+
+test("Quantity is populated correctly for non dispenses", () => {
+  dispenseFormValues.lineItems[0].statusCode = LineItemStatus.TO_BE_DISPENSED
+  dispenseFormValues.lineItems[0].suppliedQuantityValue = "1"
+
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  const resultMedicationDispense = getMedicationDispenseResources(result)
+
+  expect(resultMedicationDispense[0].quantity.value).toEqual(0)
+})
+
+test("Adds repeat information extension result when prescription is continuous repeat", () => {
+  const result = createDispenseNotification(messageHeader, patient, medicationRequests, dispenseFormValues)
+  const resultMedicationDispense = getMedicationDispenseResources(result)
+  const repeatExtension =
+    resultMedicationDispense[0].extension.find(e =>
+      e.url === "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation"
+    )
+  expect(repeatExtension).toBeTruthy()
```

Comment:
generally don't love `toBeTruthy/Falsy` just because they are quite general and can hide tests that shouldn't pass, something like `toBe/Equal(xyz), toBeUndefined(), toBeNull` give a little more clarity

`repeatExtension` might be quite a complicated object but you could maybe do `expect(repeatExtension."something"."value").toEqual("some value") instead